packages/eslint-plugin/tests/rules/no-empty-interface.test.ts

```diff
@@ -75,3 +77,63 @@ interface Bar extends Foo {}
     },
   ],
 });
+
+describe('no-empty-interface | "noEmptyWithSuper" fixer', () => {
+  const linter = new TSESLint.Linter();
+  linter.defineRule('no-empty-interface', rule);
+  linter.defineParser('@typescript-eslint/parser', parser);
+
+  function testOutput(
+    code: string,
+    output: string,
+    allowSingleExtends = false,
+  ) {
+    it(code, () => {
+      const result = linter.verifyAndFix(
+        code,
+        {
+          rules: { 'no-empty-interface': [2, { allowSingleExtends }] },
+          parser: '@typescript-eslint/parser',
+        },
+        { fix: true },
+      );
+
+      expect(result.messages).toHaveLength(0);
+      expect(result.output).toBe(output);
+    });
+  }
+
+  testOutput(
+    'interface Foo extends Array<number> {}',
+    'type Foo = Array<number>',
+  );
+
+  testOutput(
+    'interface Foo extends Array<number | {}> { }',
+    'type Foo = Array<number | {}>',
+  );
+
+  testOutput(
+    `
+interface Bar {
+  bar: string;
+}
+interface Foo extends Array<Bar> {}
+`,
+    `
+interface Bar {
+  bar: string;
+}
+type Foo = Array<Bar>
+`,
+  );
+
+  testOutput(
+    `
+type R = Record<string, unknown>;
+interface Foo extends R {   };`,
+    `
+type R = Record<string, unknown>;
+type Foo = R;`,
+  );
+});
```

Comment:
You should be able to test this fixer using the existing eslint rule tester above.

You can add an invalid case like this, and eslint will automatically test the fixer:

```ts
{
  code: `
type R = Record<string, unknown>;
interface Foo extends R {   };
  `,
  output: `
type R = Record<string, unknown>;
type Foo = R;
  `,
  options: [{ allowSingleExtends: false }],
  errors: [
    {
      messageId: 'noEmptyWithSuper',
      line: 3,
      column: 0,
    },
  ],
}
```

Reply:
Super simple, thanks for the right direction.